web/components/roomConfig/RoomConfig.tsx

```diff
@@ -0,0 +1,124 @@
+        })
+        .catch(ex => {
+            console.warn("Failed to fetch existing connections", ex);
+            setError("Failed to fetch existing connections");
+        });
+    }
+
+    if (serviceConfig === null) {
+        api.getServiceConfig<SConfig>(type)
+        .then((res) => setServiceConfig(res))
+        .catch(ex => {
+            console.warn("Failed to fetch service config", ex);
+            setError("Failed to fetch service config");
+        });
+
+    }
+
+    // We need to increment this every time we create a connection in order to properly reset the state.
+    const [ newConnectionKey, setNewConnectionKey ] = useState<number>(0);
+
+    return <>
+        <main>
+            {
+                error && <ErrorPane header="Error">{error}</ErrorPane>
+            }
+            <header className={style.header}>
+                <img src={headerImg} />
+                <h1>{text.header}</h1> 
+            </header>
+            { canEditRoom && <section>
+                <h2>{text.createNew}</h2>
+                {serviceConfig && <ConnectionConfigComponent
+                    key={newConnectionKey}
+                    api={api}
+                    serviceConfig={serviceConfig}
+                    onSave={(config) => {
+                        api.createConnection(roomId, connectionEventType, config).then(() => {
+                            // Force reload
+                            setConnections(null);
+                            setNewConnectionKey(newConnectionKey+1);
+                        }).catch(ex => {
+                            console.warn("Failed to create connection", ex);
+                            setError("Failed to create connection");
+                        });
+                    }}
+                />}
+            </section>}
+            <section>
+                <h2>{ canEditRoom ? text.listCanEdit : text.listCantEdit }</h2>
+                { serviceConfig && connections?.map(c => <ListItem key={c.id} text={listItemName(c)}>
+                        <ConnectionConfigComponent
+                            api={api}
+                            serviceConfig={serviceConfig}
+                            existingConnection={c}
+                            onSave={(config) => {
+                                api.updateConnection(roomId, c.id, config).then(() => {
+                                    // Force reload
+                                    setConnections(null);
+                                }).catch(ex => {
+                                    console.warn("Failed to create connection", ex);
+                                    setError("Failed to create connection");
+                                });
+                            }}
+                            onRemove={() => {
+                                api.removeConnection(roomId, c.id).then(() => {
+                                    setConnections(connections.filter(conn => c.id !== conn.id));
```

Comment:
Async update using a potentially outdated `connections` :warning: 
```suggestion
                                    setConnections(connections => connections.filter(conn => c.id !== conn.id));
```